ddtrace/opentracer/tracer.go

```diff
@@ -84,3 +87,12 @@ func (t *opentracer) Extract(format interface{}, carrier interface{}) (opentraci
 		return nil, opentracing.ErrUnsupportedFormat
 	}
 }
+
+// ContextWithSpan implements opentracing.TracerContextWithSpanExtension.
+func (t *opentracer) ContextWithSpanHook(ctx context.Context, openSpan opentracing.Span) context.Context {
+	ddSpan, ok := openSpan.(*span)
```

Comment:
It might be better to rely on public interfaces, this variant should work

```suggestion
	ddSpan, ok := interface{}(openSpan).(ddtrace.Span)
	if !ok {
		return ctx
	}
	return tracer.ContextWithSpan(ctx, ddSpan)
```

Reply:
That's a very weird syntax I haven't seen before. I guess you live and you learn 🙂 Why would this be better? I think the previous version was very clear and explicit, this one seem a bit confusing (to me), but I'm open.

Reply:
It’s looks a bit weird because it consists of two operations: cast to ‘interface{}’ (it’s needed due to method names collisions) and then type assertion to ‘ddtrace.Span’ (because underlining struct implements this interface).
In other words we just checking if type that comes as ‘opentracing.Span’ also implements ‘ddtrace.Span’.

Reply:
Thanks for the explanation. I get what it does, I just think that the code as it is now is clearer and simpler. I'd prefer we don't change it.